utilities/blob_db/blob_compaction_filter.h

```diff
@@ -11,55 +11,72 @@
 #include "monitoring/statistics.h"
 #include "rocksdb/compaction_filter.h"
 #include "rocksdb/env.h"
+#include "rocksdb/utilities/layered_compaction_filter_base.h"
 #include "utilities/blob_db/blob_db_gc_stats.h"
 #include "utilities/blob_db/blob_db_impl.h"
 
 namespace ROCKSDB_NAMESPACE {
 namespace blob_db {
 
 struct BlobCompactionContext {
+  BlobDBImpl* blob_db_impl = nullptr;
   uint64_t next_file_number = 0;
   std::unordered_set<uint64_t> current_blob_files;
   SequenceNumber fifo_eviction_seq = 0;
   uint64_t evict_expiration_up_to = 0;
 };
 
 struct BlobCompactionContextGC {
-  BlobDBImpl* blob_db_impl = nullptr;
   uint64_t cutoff_file_number = 0;
 };
 
 // Compaction filter that deletes expired blob indexes from the base DB.
 // Comes into two varieties, one for the non-GC case and one for the GC case.
-class BlobIndexCompactionFilterBase : public CompactionFilter {
+class BlobIndexCompactionFilterBase : public LayeredCompactionFilterBase {
  public:
-  BlobIndexCompactionFilterBase(BlobCompactionContext&& context,
-                                uint64_t current_time, Statistics* stats)
-      : context_(std::move(context)),
+  BlobIndexCompactionFilterBase(
+      BlobCompactionContext&& blob_comp_context,
```

Comment:
Again, `context` would be preferred, here and in the other class below.

Reply:
Change to context would have build warnings, as we have context() in BlobIndexCompactionFilterBase. Let me use _context instead.